### PR TITLE
feat(FEC-8046): 360 support

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* expose VR API's and event
* enable to pass options to event target (needed to pass `passive` by 'vr' plugin) 
* enrich the Dom util

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1205,6 +1205,27 @@ export default class Player extends FakeEventTarget {
...
+  // </editor-fold>
+
+  // <editor-fold desc="Plugins API">
+
+  /**
+   * get a plugin instance
+   * @param {string} name - the plugin name
+   * @returns {?BasePlugin} - the plugin instance
+   */
+  getPlugin(name: string): ?BasePlugin {
```
Comment (dan-ziv) [inline src/player.js]: If we are already adding this API we need to add also something like
```js
isPluginActive(name) : boolean
```

File: src/error/category.js [hunk 2]
```diff
@@ -28,7 +28,10 @@ const Category: CategoryType = {
   'ADS': 8,
 
   /** Errors in the database storage (offline). */
-  'STORAGE': 9
+  'STORAGE': 9,
+
+  /** Errors from a plugin. */
+  'PLUGIN': 11
```
Comment (OrenMe) [inline src/error/category.js]: no need for plugin name-spacing, too generic to have the  `PLUGIN_FATAL_ERROR`, we already have a fatal error convention via the `SEVERITY` so it's also redundant to create an event with `FATAL` naming IMO.

File: src/player.js (hunk 1 above)
Comment (OrenMe) [reply src/player.js]: I see a case of exposing the plugins outside, think of cases where we are also using a 3rd party API that might be needed for application integration(hls.js needed by a 3rd party we don't implement, IMA is needed for MOAT integration etc.)

As we are adding VR specific stuff I would like to keep it focused and not add unplanned APIs.
@yairans add the `toggleVrStereoMode` here and refer it to the plugin directly - via the plugin manager, and change the UI to call `player.toggleVrStereoMode()`.
You can also remove the kaltura player bindings then.
Once we create the controller for VR we will change this implementation to use it.

Comment (OrenMe) [reply src/player.js]: also, maybe toggle is not the best option, as it forces you to know the state, so you will have to also expose `isStereoModeSet`

Comment (yairans) [reply src/player.js]: done

**Pull request merged**